src/decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super(nameable)
```

Comment:
- [ ] [OPTIONAL] In this case you shouldn't pass nameable to the `super` method. the superclass `Nameable` does not have a constructor that accepts arguments.